cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -939,4 +953,23 @@ private String queryBySql(boolean hasIndex, IndexTextAndJsonValues vals, CQLTerm
     return sql;
   }
 
+  private String appendFullTermIfTooLong(String comparator, String index, String termUnmod, String sql) {
+    if(termUnmod.length() >= 600) {
```

Comment:
Can we be sure that for all unicode strings Java's length calculation and PostgreSQL's length calculation is always the same?
If not this if should be converted into SQL.

Reply:
ok so i spent some time looking at this and I agree with the need to sync up the length measurements.  Because the string length of Java could not match what the database is using.  Couple questions I have about this.  

My first is what the return value would be for a single function to handle any query.  RETURNS SETOF XXXXX.  What should XXXXX be?  It would have to be parameterized to the table its called on?  I dont see a way to do that.  

Second I would assume I would have to send in the query and execute it the same way I am doing in Java.  This would essentially be replacing parts of the java we have with sql.

Reply:
I don't see the need for a function. This is an example query. 'foo' will be replaced by the search string.
```
SELECT * FROM t WHERE 
    left(jsonb->>'title', 600) = left('foo', 600) 
    AND CASE WHEN length('foo') > 600
             THEN jsonb->>'title' = 'foo' 
             ELSE TRUE
        END;
```

Reply:
If you want to create a function you need to create a separate function for each field where an b-tree index is created. Example:
```
CREATE OR REPLACE FUNCTION titlelike(instance, text) RETURNS boolean AS $$
  SELECT left($1.jsonb->>'title', 600) LIKE left($2, 600)
    AND CASE WHEN length($2) > 600
             THEN  $1.jsonb->>'title' LIKE $2
             ELSE TRUE END;
$$ LANGUAGE SQL;
```

Reply:
I believe I have resolved this.  Will work through failures now